### PR TITLE
[Modules] New module for `Microsoft.ResourceGraph/queries`

### LINKED ISSUE #2184: [Modules] Create new `Microsoft.ResourceGraph/queries` module
Create a new CARML module for the `Microsoft.ResourceGraph/queries` resource type as per the [service documentation](https://learn.microsoft.com/en-us/azure/templates/microsoft.resourcegraph/queries?pivots=deployment-language-bicep) and our [contribution guide](https://github.com/Azure/ResourceModules/wiki/Contribution%20guide).

Examples values could be:

- queryCode: `'Resources | where type =~ 'Microsoft.Compute/virtualMachines' | summarize count() by tostring(properties.storageProfile.osDisk.osType)'`
- queryDescription: `'This shared query counts all virtual machine resources and summarizes by the OS type.'`

### PR DESCRIPTION
**Pull request by krbar**
# Description

This PR resolves #2184 and adds a new CARML module for `Microsoft.ResourceGraph/queries`.

## Pipeline references

| Pipeline |
| - |
| [Build Status](https://dev.azure.com/carml/CARML/_build/latest?definitionId=136&branchName=users%2Fkrbar%2F2184-resourcegraphqueries-module) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/resource-graph/query/.test/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,15 @@
+targetScope = 'subscription'
+
+// ========== //
+// Test Cases //
+// ========== //
+
+var namePrefix = 'crml'
+
+// TEST 1 - COMMON
```
Comment (AlexanderSehr) [inline modules/resource-graph/query/.test/main.test.bicep:9]: While it seams over the top, please let's also a min test.... Which I guess will be very small 😄

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

File: modules/resource-graph/query/.test/main.test.bicep (hunk 1 above)
Comment (krbar) [reply modules/resource-graph/query/.test/main.test.bicep:9]: Min test added.

File: modules/resource-graph/query/.test/min/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.resourcegraph.queries-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'rgqcom'
```
Comment (AlexanderSehr) [inline modules/resource-graph/query/.test/min/main.test.bicep:15]:
```suggestion
param serviceShort string = 'rgqmin'
```

File: modules/resource-graph/query/main.bicep [hunk 3]
```diff
@@ -0,0 +1,88 @@
+metadata name = 'Resource Graph Queries'
+metadata description = 'This module deploys a Resource Graph Query.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Optional. Name of the Resource Graph Query.')
+param name string = guid(resourceGroup().id)
```
Comment (AlexanderSehr) [inline modules/resource-graph/query/main.bicep:6]: Is this default a good idea? Wouldn't that mean you can only deploy a single query per resource group if using the default?

File: modules/resource-graph/query/main.bicep [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    principalIds: roleAssignment.principalIds
+    principalType: contains(roleAssignment, 'principalType') ? roleAssignment.principalType : ''
+    roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
+    condition: contains(roleAssignment, 'condition') ? roleAssignment.condition : ''
+    delegatedManagedIdentityResourceId: contains(roleAssignment, 'delegatedManagedIdentityResourceId') ? roleAssignment.delegatedManagedIdentityResourceId : ''
+    resourceId: rgQuery.id
+  }
+}]
+
+@description('The name of the user assigned identity.')
```
Comment (AlexanderSehr) [inline modules/resource-graph/query/main.bicep:78]:
```suggestion
@description('The name of the query.')
```

File: modules/resource-graph/query/main.bicep [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+output name string = rgQuery.name
+
+@description('The resource ID of the user assigned identity.')
```
Comment (AlexanderSehr) [inline modules/resource-graph/query/main.bicep:81]:
```suggestion
@description('The resource ID of the query.')
```

File: modules/resource-graph/query/main.bicep [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+output resourceId string = rgQuery.id
+
+@description('The resource group the user assigned identity was deployed into.')
```
Comment (AlexanderSehr) [inline modules/resource-graph/query/main.bicep:84]:
```suggestion
@description('The resource group the query was deployed into.')
```

File: modules/resource-graph/query/main.bicep (hunk 3 above)
Comment (krbar) [reply modules/resource-graph/query/main.bicep:6]: Definitely not. This is not intentional, copy/paste error, sorry about that...

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**